File: packages/js-sdk/src/regex/detectors/pl/nip.ts
// Polish NIP (tax identification number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { plNipChecksum } from '../../validators'

class PlNipDetector extends RegexDetector {
  entityType = EntityType.PL_NIP
  score = 0.85
  contextKeywords = [
    'nip', 'tax', 'podatkowy',
  ]
  contextRequired = true

  pattern = /\b\d{3}[-\s]?\d{3}[-\s]?\d{2}[-\s]?\d{2}\b/g
  validator = plNipChecksum
}

registerRegion('pl', PlNipDetector)

export { PlNipDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/pl/nip.py
"""Polish NIP (Tax Identification Number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import pl_nip_checksum


@register_region("pl")
class NipDetector(RegexDetector):
    entity_type = EntityType.PL_NIP
    score = 0.85
    context_keywords = [
        "nip", "tax", "podatkowy",
    ]
    context_required = True
    context_window = 50

    # 10 digits with optional separators
    pattern = re.compile(
        r"\b\d{3}[-\s]?\d{3}[-\s]?\d{2}[-\s]?\d{2}\b"
    )
    validator = staticmethod(pl_nip_checksum)
